src/qt/automintmodel.h

```diff
@@ -111,4 +112,67 @@ public Q_SLOTS:
     IncomingFundNotifier *notifier;
 };
 
+enum class AutoMintSparkState : uint8_t {
+    Disabled,
+    WaitingIncomingFund,
+    WaitingUserToActivate,
+    Anonymizing
+};
+
+enum class AutoMintSparkAck : uint8_t {
+    AskToMint,
+    Success,
+    WaitUserToActive,
+    FailToMint,
+    NotEnoughFund,
+    UserReject,
+    FailToUnlock,
+    Close
+};
+
+class AutoMintSparkModel : public QObject
```

Comment:
Again, why we need separate types and classes, we can just reused what we already have,